shaka/src/mapping/v8/js_wrappers.cc

```diff
@@ -238,18 +238,19 @@ bool IsDerivedFrom(BackingObject* ptr, const std::string& name) {
 
 bool RunScript(const std::string& path) {
   util::FileSystem fs;
-  std::vector<uint8_t> source;
-  CHECK(fs.ReadFile(path, &source));
+  std::string source;
+  CHECK(fs.ReadFile(path, source));
   v8::Local<v8::String> code =
       v8::String::NewFromUtf8(GetIsolate(),
-                              reinterpret_cast<const char*>(source.data()),
-                              v8::NewStringType::kNormal, source.size())
+                              source.c_str(),
+                              v8::NewStringType::kNormal
+                              source.length())
           .ToLocalChecked();
   return RunScriptImpl(path, code);
 }
 
-bool RunScript(const std::string& path, const uint8_t* data, size_t data_size) {
-  v8::Local<v8::String> source = MakeExternalString(data, data_size);
+bool RunScript(const std::string& path, const std::string& data) {
+  v8::Local<v8::String> source = MakeExternalString(data.c_str(), data.length());
```

Comment:
Unfortunately I don't think this will work.  The reason the original method accepted a `uint8_t*` was because the "external" strings avoids a copy.  But I think the pointer needs to live beyond the return of this method.  So you should just use `JsStringFromUtf8` to make a copy.